src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -91,6 +91,14 @@ def _run(self, verbose, **job_kwargs):
         progress_bar = job_kwargs["progress_bar"]
 
         unit_ids = self.sorting.unit_ids
+        non_empty_unit_ids = self.sorting.get_non_empty_unit_ids()
+        empty_unit_ids = self.sorting.get_empty_unit_ids()
```

Comment:
Counting spikes have a cost.
Here we are doing it twice!
And then later we are doing "compute_num_spikes".
So we are counting how many spikes 3 times in this functions.
I think once should be enough

Reply:
done!